### ISSUE TITLE
detach() functions in G and D optimization

### ISSUE THREAD

**Issue by sdlpkxd** [xinntao/BasicSR#233]
Excuse me. Thanks for your great work. I have a question.

In optimize_parameters() function in SRGAN_model.py, I find the pytorch detach() function is used for G and D seperately. In D optimization, self.fake_H.detach()is used to avoid BP to G, because the fake_H is generated by generation net. But why there are detach() operations for var_ref and var_H (self.netF(self.var_H).detach() or self.netD(self.var_ref).detach()) in G optimization, they should not be used for BP ?

Thanks in advance.
